File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        unordered_map<int, int> seen;
        seen[0] = 1;
        int subCnt = 0;
        int prefixsum = 0;
        for (int i : nums) {
            prefixsum += i;
            int req = prefixsum - k;
            if (seen.find(req) != seen.end()) {
                subCnt += seen[req];
            }
            seen[prefixsum]++;
        }
        return subCnt;
        
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        seen = {0:1}
        subCnt=0
        prefixsum=0
        for i in nums:
            prefixsum+=i
            req=prefixsum-k
            if req in seen :
                subCnt+=seen[req]
            seen[prefixsum]=seen.get(prefixsum,0)+1
        return subCnt